storage/badger/chunkDataPacks.go

```diff
@@ -30,9 +30,9 @@ func NewChunkDataPacks(collector module.CacheMetrics, db *badger.DB, collections
 	retrieve := func(key interface{}) func(tx *badger.Txn) (interface{}, error) {
 		chunkID := key.(flow.Identifier)
 
-		var c *badgermodel.StoredChunkDataPack
+		var c badgermodel.StoredChunkDataPack
 		return func(tx *badger.Txn) (interface{}, error) {
-			err := operation.RetrieveChunkDataPack(chunkID, c)(tx)
+			err := operation.RetrieveChunkDataPack(chunkID, &c)(tx)
 			return &c, err
```

Comment:
returning a double reference, i.e., `**badgermodel.StoredChunkDataPack` was the root cause, but since tests were reading from the storage, this function had never been tested. I added tests covering this function. So such type assertion errors are being caught.

Reply:
Hmm, I can't see how this could cause problem. Care to elaborate a bit more?
If we store double pointer, it should fail cast to single pointer we use in a few places.
Cache is also keyed by identifier.

Reply:
>Hmm, I can't see how the could cause problem. Care to elaborate a bit more?
If we store double pointer, it should fail cast to single pointer we use in a few places.
Cache is also keyed by identifier.

@m4ksio the `retrieve` function of `ChunkDataPacks` is only called when there is a [cache miss and the storage is queried directly](https://github.com/onflow/flow-go/blob/f33503f3362a009fadf9783fd141309242a463aa/storage/badger/cache.go#L94-L101). We did not _store_ double-pointer. However, we tried _retrieving_ a single pointer as a double-pointer, which would fail the type assertion. Also, this issue was only confined to the `retrieve` function for `ChunkDataPacks`.
The cache has a separate path on [store](https://github.com/onflow/flow-go/blob/f33503f3362a009fadf9783fd141309242a463aa/storage/badger/cache.go#L119-L126) and [retrieval](https://github.com/onflow/flow-go/blob/f33503f3362a009fadf9783fd141309242a463aa/storage/badger/cache.go#L87-L92) which is completely independent of `retrieve`.  

To prove this hypothesis, you can perform the following experiment on _this branch_:
- Revert `retrieve` function to return a double pointer.
- Run [TestChunkDataPacks_Store](https://github.com/onflow/flow-go/blob/021346334f0444f7fd9f95523b7d64a88fd03e78/storage/badger/chunk_data_pack_test.go#L21-L44) -> it fails with this error: `could not retrive stored chunk data pack: could not retrieve resource: could not decode entity: msgpack: Decode(nonsettable *badgermodel.StoredChunkDataPack)`, which is the same error we saw on benchnet. This is basically since the cache size was set to 1 so 99 out of 100 queries are routed to the storage. Previously, the cache size was big enough that all queries would be made through the cache.
- Now bump up the cache size of `ChunkDataPacks` to 100 and the test is passed. Since we have 100 chunk data packs and the cache is big enough to answer all queries!